### PR TITLE
feat(workspace/app_server): add resource to manage server

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new resource to manage server under specified server group.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to manage a server.
2. add corresponding document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
./scripts/coverage.sh -o workspace -f TestAccResourceAppServer_
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccResourceAppServer_ -timeout 360m -parallel 10
=== RUN   TestAccResourceAppServer_basic
=== PAUSE TestAccResourceAppServer_basic
=== RUN   TestAccResourceAppServer_prepaid
=== PAUSE TestAccResourceAppServer_prepaid
=== CONT  TestAccResourceAppServer_basic
=== CONT  TestAccResourceAppServer_prepaid
--- PASS: TestAccResourceAppServer_basic (719.19s)
--- PASS: TestAccResourceAppServer_prepaid (1387.24s)
PASS
coverage: 12.2% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1387.303s       coverage: 12.2% of statements in ./huaweicloud/services/workspace
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found    
image
    
    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
     When the resource does not exist, the response status code is always 200.

    
    bb. Related resources (parent resources) not found
    When the Workspace service or cloud application is disabled.
image

### CODE REVIEW

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_app_server.go [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+	}
+
+	server := utils.PathSearch("items|[0]", respBody, nil)
+	if server == nil {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return server, nil
+}
+
+func resourceAppServerUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_app_server.go:458]: The update method for the field 'auto_renew' seems missing.

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o workspace -f TestAccResourceAppServer_
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccResourceAppServer_ -timeout 360m -parallel 10
=== RUN   TestAccResourceAppServer_basic
=== PAUSE TestAccResourceAppServer_basic
=== RUN   TestAccResourceAppServer_prepaid
=== PAUSE TestAccResourceAppServer_prepaid
=== CONT  TestAccResourceAppServer_basic
=== CONT  TestAccResourceAppServer_prepaid
--- PASS: TestAccResourceAppServer_basic (792.31s)
--- PASS: TestAccResourceAppServer_prepaid (906.33s)
PASS
coverage: 11.0% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 906.458s        coverage: 11.0% of statements in ./huaweicloud/services/workspace

File: docs/resources/workspace_app_server.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
+---
+subcategory: "Workspace"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_workspace_app_server"
+description: |-
+  Manages a Workspace APP server resource under specified server group within HuaweiCloud.
+---
+
+
```
Comment (Lance52259) [inline docs/resources/workspace_app_server.md]:
```suggestion

```
extra line needs watch out

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_app_server.go [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+	if periodUnit == "month" {
+		return 2
+	}
+
+	if periodUnit == "year" {
+		return 3
+	}
+
+	return nil
+}
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_app_server.go]:
```suggestion
func buildCreatePeriodTypeParam(periodUnit string) interface{} {
	switch periodUnit {
	case "month":
		return 2
	case "year":
		return 3
	}
	return nil
}
```

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_app_server.go [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+	}
+
+	return resourceAppServerRead(ctx, d, meta)
+}
+
+func waitForPrePaidServerComplete(ctx context.Context, cfg *config.Config, region string, timeOut time.Duration, resp interface{}) (string, error) {
+	orderId := utils.PathSearch("order_id", resp, "").(string)
+	if orderId == "" {
+		return "", fmt.Errorf("unable to find order ID from API response")
+	}
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_app_server.go]:
```suggestion
func waitForPrePaidServerComplete(ctx context.Context, cfg *config.Config, region string, timeOut time.Duration, orderId string) (string, error) {
	if orderId == "" {
		return "", fmt.Errorf("order ID of prePaid server is empty)
	}
```

**Comment by Lance52259** [comment]
title perfer: feat(workspace/app): add resource to manage server

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccResourceAppServer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccResourceAppServer_ -timeout 360m -parallel 4
=== RUN   TestAccResourceAppServer_basic
=== PAUSE TestAccResourceAppServer_basic
=== RUN   TestAccResourceAppServer_prepaid
=== PAUSE TestAccResourceAppServer_prepaid
=== CONT  TestAccResourceAppServer_basic
=== CONT  TestAccResourceAppServer_prepaid
--- PASS: TestAccResourceAppServer_basic (767.17s)
--- PASS: TestAccResourceAppServer_prepaid (879.67s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 879.786s

**Pull request merged**